nanomq/sub_handler.c

```diff
@@ -336,15 +275,20 @@ sub_ctx_handle(nano_work *work)
 			}
 			tq = tq->next;
 		}
-		if (!topic_exist && topic_str) {
-			// printf("Protocol Version: %d\n",
-			// old_ctx->proto_ver); printf("sub ID: %d\n",
-			// work->sub_pkt->sub_id.varint);
-			int          t = work->sub_pkt->sub_id.varint;
+		if (!topic_exist) {
+			int            t = 0;
+			if (conn_param_get_protover(work->cparam) ==
+			    PROTOCOL_VERSION_v5) {
+				property_data *pdata = property_get_value(
+				    work->sub_pkt->properties,
+				    SUBSCRIPTION_IDENTIFIER);
+				if (pdata) {
+					t = pdata->p_value.varint;
+				}
+			}
 			dbtree_ctxt *db_old_ctxt = dbtree_new_ctxt(old_ctx, t);
 			dbtree_insert_client(
 			    work->db, topic_str, db_old_ctxt, work->pid.id);
-
 			dbhash_insert_topic(work->pid.id, topic_str);
 		}
 
```

Comment:
Here we need to reset variable ```r``` to NULL.

Reply:
`r` ?